### PR TITLE
Added option to show Map instead of Gear in MISC

### PR DESCRIPTION
**Pull request by photon1503**
Hi Andreas,

this PR adds the option to show a Minimap instead of the Gear. 

Best regards,
Gerald

### CODE REVIEW

**Comment by photon1503** [comment]
Update:
* Changed design of MISC tab to 2-column grid for the options to save space.
* Added option to override automatic selection of dashboard type

**Comment by andreasdahl1987** [comment]
Hi Gerald!

Thank you for this! 

Can you add a version of the dashboard with the map like you've visioned it?

Kind regards,
Andreas

**Comment by andreasdahl1987** [comment]
Just built the changes, great improvements. The option to chose dashboard type is a very nice addition :)

File: Dahl Design Properties.sln [hunk 1]
```diff
@@ -1,10 +1,15 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.25123.0
```
Comment (mcowger) [inline Dahl Design Properties.sln:4]: I'd suggest excluding your changes to the solution definition from the commit

File: DataPluginDemo.cs [hunk 2]
```diff
@@ -1379,7 +1387,7 @@ public void DataUpdate(PluginManager pluginManager, ref GameData data)
                     pluginManager.SetPropertyValue("IdleRPM", this.GetType(), idleRPM);
 
                     pluginManager.SetPropertyValue("CenterDashType", this.GetType(), dashType);
-                    pluginManager.SetPropertyValue("MenuType", this.GetType(), rotaryType);
+                    pluginManager.SetPropertyValue("MenuType", this.GetType(), rotaryType);
```
Comment (mcowger) [inline DataPluginDemo.cs:1390]: Remove whitespace at EOL

File: DataPluginDemo.cs [hunk 3]
```diff
@@ -6916,6 +6924,8 @@ public void Init(PluginManager pluginManager)
             pluginManager.AddProperty("DDUEnabled", this.GetType(), Settings.DDUEnabled);
             pluginManager.AddProperty("SW1Enabled", this.GetType(), Settings.SW1Enabled);
             pluginManager.AddProperty("DashLEDEnabled", this.GetType(), Settings.DashLEDEnabled);
+            pluginManager.AddProperty("ShowMap", this.GetType(), Settings.ShowMap);
```
Comment (mcowger) [inline DataPluginDemo.cs:6927]: This seems like a boolean - maybe `ShowMapEnabled`

File: DataPluginDemo.cs [hunk 4]
```diff
@@ -7643,9 +7653,7 @@ public void Init(PluginManager pluginManager)
             pluginManager.AddProperty("SW1Clutch", this.GetType(), 0);
             pluginManager.AddProperty("SW1BitePoint", this.GetType(), 0);
             pluginManager.AddProperty("SW1Brake", this.GetType(), 0);
-            pluginManager.AddProperty("SW1Throttle", this.GetType(), 0);
-
```
Comment (mcowger) [inline DataPluginDemo.cs:7647]: Remove whitespace.

File: SettingsControlDemo.xaml [hunk 5]
```diff
@@ -249,14 +249,14 @@
                     <styles:SHSection Title="Miscellaneous" Margin="0,0,0,0">
                         <StackPanel>
                             <TextBlock
-                                Text="Fuel per lap target:" Grid.Row="1" Margin="10,25,0,23" Grid.RowSpan="2"/>
+                                Text="Fuel per lap target:" Grid.Row="1" Margin="10,25,0,23"/>
```
Comment (mcowger) [inline SettingsControlDemo.xaml:252]: These dont seem relevant to the map display change...

File: SettingsControlDemo.xaml [hunk 6]
```diff
@@ -249,14 +249,14 @@
...
                             <TextBox
                                 VerticalAlignment="Top"
                                 Height="20"  
                                 Margin="10,-20,0,-20"                        
                                 Text="{Binding fuelPerLapTarget}"/>
                             <TextBlock
-                                Text="Fuel per lap offset increments:" Grid.Row="1" Margin="10,10,0,23" Grid.RowSpan="2"/>
+                                Text="Fuel per lap offset increments:" Grid.Row="1" Margin="10,10,0,23" />
```
Comment (mcowger) [inline SettingsControlDemo.xaml:259]: These dont seem relevant to the map display change...

File: install.cmd [hunk 7]
```diff
@@ -0,0 +1,2 @@
+copy bin\Release\DahlDesign.dll "%PROGRAMFILES(X86)%\SimHub\"
```
Comment (mcowger) [inline install.cmd:1]: the build target will do this for you...

**Comment by mcowger** [comment]
Made some comments, most are just nits.

**Comment by mcowger** [comment]
Would be interesting to find a way to keep the minimap (which I like) but also find a place to show the keep (maybe overlaid on the RPM bar)...thoughts @photon1503 ?

**Comment by photon1503** [comment]
> Would be interesting to find a way to keep the minimap (which I like) but also find a place to show the keep (maybe overlaid on the RPM bar)...thoughts @photon1503 ?

Thank you for the review and the comments. Since I use a dedicated device for the current gear, there was no need to show it on the dashboard.  But I am open for any ideas :-)

File: DataPluginDemo.cs (hunk 3 above)
Comment (photon1503) [reply DataPluginDemo.cs:6927]: I will rename it accordingly

**Comment by photon1503** [comment]
> ersion of the dashboard with the ma



> Can you add a version of the dashboard with the map like you've visioned it?

There is already an PR https://github.com/andreasdahl1987/DahlDesignDash/pull/2

**Comment by andreasdahl1987** [comment]
> > Can you add a version of the dashboard with the map like you've visioned it?
> 
> There is already an PR [andreasdahl1987/DahlDesignDash#2](https://github.com/andreasdahl1987/DahlDesignDash/pull/2)

Ah man, I'm not paying attention 👀

**Comment by andreasdahl1987** [comment]
> > Would be interesting to find a way to keep the minimap (which I like) but also find a place to show the keep (maybe overlaid on the RPM bar)...thoughts @photon1503 ?
> 
> Thank you for the review and the comments. Since I use a dedicated device for the current gear, there was no need to show it on the dashboard. But I am open for any ideas :-)

I was suspecting something like this. I know quite a few people have a LED matrix showing gear number next to the display. So I think this option is a good addition. Maybe a bit niche, but still good. 

On that subject, I'm planning to extract the many pages of the left and right info screens out of the widget pages, and instead have them all on one page with folders controlled by a plugin property for visibility. This would make it possible to do a few things:
- Have plugin specific bindings for the pages, instead of using simhub Action A, B, C, D
- Make the dash automatically open qualitying pages for qualifyers, race pages for races etc.
- Options to select your preferred pages for specific sessions

**Comment by photon1503** [comment]
Andreas, your assumption was perfectly right: 
image
 
Your plan to refactor the pages on the Dashboard sounds good.

My next step was to add an option to choose the "default" page for the left and right info screen. But I already see this on your TODO list :-)
In the previous version I had this hardcoded directly in your dashboard but I would like to see this as an feature. Because currently I always need to click "Action A" a few times until my preferred screen is shown.

**Comment by andreasdahl1987** [comment]
> Andreas, your assumption was perfectly right: image
> 
> Your plan to refactor the pages on the Dashboard sounds good.
> 
> My next step was to add an option to choose the "default" page for the left and right info screen. But I already see this on your TODO list :-) In the previous version I had this hardcoded directly in your dashboard but I would like to see this as an feature. Because currently I always need to click "Action A" a few times until my preferred screen is shown.

Looking at the way you set it up with the gear pages I think Ive got an idea of how to organize it! 

After #5 there are some conflicts with the .xaml file. If you could sort those out Ill merge

**Review by mcowger — COMMENTED** [review]
LGTM!

**Comment by photon1503** [comment]
I updated SettingsControlDemo.xaml and inserted CoupleInCarToPit into the new Grid style view. Please check if it's ok now. Sorry for the many commits, this is my first PR ever :-)

**Comment by mcowger** [comment]
For a first PR its REALLY good :).   Well done!

You might consider [squashing commits](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits) for the future, but its really a nit pick.

**Comment by andreasdahl1987** [comment]
@photon1503 I have never done a PR myself, and I'm all new to git. Big thanks for your contribution, and thanks @mcowger for reviewing 👏

**Comment by photon1503** [comment]
Thank you for restoring the .sln file. I also just noticed that I deleted it unintentionally, sorry

**Pull request merged**